gammapy/utils/random.py

```diff
@@ -78,16 +70,16 @@ def sample_sphere(size, lon_range=None, lat_range=None, unit='rad'):
 
     # Sample random latitude
     v = np.random.random(size)
-    z_range = np.sin(np.array(lat_range))
+    z_range = np.sin(lat_range)
     z = z_range[0] + (z_range[1] - z_range[0]) * v
     # This is not the formula given in the reference, but it is equivalent.
     lat = np.arcsin(z)
 
     # Return result
-    if unit == 'rad':
-        return lon, lat
-    elif unit == 'deg':
-        return np.degrees(lon), np.degrees(lat)
+    if lon_format == '-180_180_deg':
+        # convert back to (-180, 180) deg
+        lon -= Angle(180., 'degree')
```

Comment:
Ok, I gave it some more thought, and revised the docs about `Longitude` and `Latitude`. I problem I see is that the `Longitude` class, because of the wrapping method, is not very convenient for setting longitude ranges. For instance, when setting the range `(0, 360) deg`, it will be wrapped to `(0, 0) deg`!!! One would have to define the `2 \pi` interval as `(0, 360-epsilon) deg`, with `epsilon = 1e-8` for instance. And to remember this every time one defines the longitude interval is kind of hard (it's easy to forget).

In my opinion, `Longitude` does not make things easier (at least for the case of this function). With `Angle` you need to wrap angles, but you don't need to think about the wraping of the intrerval length. With `Longitude`, you don't need to wrap angles, but you need to remember to check if the interval is wrapped correctly.

@deil: please let me know your opinion on that.